### PR TITLE
FNOD badge fix for queue

### PR DESCRIPTION
**Pull request by bgantick**
Cleans up some naming and adds `veteran_appellant_deceased` to task column serializer

### Description
1. Cleaned up some frontend naming conventions (`veteranAppellantDeceased` vs. `veteran_appellant_deceased`)
2. Added an abstraction for `veteran_is_deceased` to the `appeal` model for readability
3. Added `veteran_appellant_deceased` to `task_column_serializer` so that attribute makes it to the frontend in task contexts

### Acceptance Criteria
- [x] Code compiles correctly
- [x] FNOD badge appears in individual queue if user has a case assigned where the appellant is a veteran and is deceased

### Testing Plan
1. Intake a deceased veteran (veteran id 45454545 should work) case (appellant should be the veteran)
2. Assign the case to an attorney (like BVASCASPER1)
3. Go to queue and make sure the newly intaken and assigned case appears with the FNOD badge next to it.

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 
Screen Shot 2021-01-11 at 12 33 47 PM

----------------

Screen Shot 2021-01-11 at 12 33 14 PM

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -342,12 +342,16 @@ def appellant_is_not_veteran
...
+    !veteran_is_not_claimant
+  end
+
   def veteran_middle_initial
     veteran_middle_name&.first
   end
 
   def veteran_appellant_deceased?
-    (veteran_is_deceased && !appellant_is_not_veteran) && FeatureToggle.enabled?(:fnod_badge, user: self)
+    FeatureToggle.enabled?(:fnod_badge, user: self) && (veteran_is_deceased && appellant_is_veteran)
```
Comment (lomky) [inline app/models/appeal.rb:354]: :green_heart:

Comment (yoomlam) [reply app/models/appeal.rb:354]: How about this?
```ruby
(veteran_is_deceased && appellant_is_veteran) if FeatureToggle.enabled?(:fnod_badge, user: self)
```

Comment (meckilabritt-navapbc) [reply app/models/appeal.rb]: Haven't since this before! Does this help performance?

Comment (lomky) [reply app/models/appeal.rb]: Yeah! in either case:

`&&`
this function won't bother to evalutate the second clause if the first returns false, so you always want to put the faster part of your two options first. 
This is also true for `||` if the first arg evaluates to true
see https://en.wikipedia.org/wiki/Short-circuit_evaluation!

`if` 
functionally works out the same as above (only do the others if this one succeeds), but reads a lot clearer as a 'dependency', at least to me!

Comment (lomky) [reply app/models/appeal.rb]: the `appellant_is_veteran` one doesn't affect performance* I just find it hard to read `!thing_not_thing`


*technically, super minutely, we do have to place a thing onto the stack and then pop it, but the time scale is so minute compared to a Database query or Network request we can consider it zero affect

**Comment by meckilabritt-navapbc** [comment]
This LGTM. Plus 1 for the performance tweak

**Review by mkerns355 — APPROVED** [review]

**Review by meckilabritt-navapbc — APPROVED** [review]

**Pull request merged**